src/Chapter2/Pagination.hs

```diff
@@ -0,0 +1,70 @@
+module Chapter2.Pagination where
+import Data.Char
+import Data.List
+
+totalPage :: TotalItems -> Int -> TotalPages
+totalPage ti ip = if (ti `mod` ip == 0)
+    then (ti `div` ip)
+    else (ti `div` ip)+1
+
+type ItemsPerPage = Int
+type TotalItems = Int
+type TotalPages = Int
+type CurrentPage = Int
+type FirstPage = Int
+type LastPage = Int
+type NumOfPagesToDisplay = Int
+
+getString :: FirstPage -> LastPage -> CurrentPage -> String
+getString fp lp cp = let
+    first = intercalate "|" (map (\s -> show s) [fp..(cp-1)])
+    current = "|" ++ (show cp) ++ "*|"
+    second = intercalate "|" (map (\s -> show s) [(cp+1)..lp])
+    in first ++ current ++ second
+
+formatPagination :: FirstPage -> LastPage -> CurrentPage  -> (Bool,Bool) -> String
```

Comment:
the pagination being rendered by this function is not in the same format as given in the problem statement. If you match the format properly, intercalation well become even easier.